pod.go

```diff
@@ -795,14 +790,17 @@ func (p *Pod) start() error {
 		return err
 	}
 
-	for _, c := range p.containers {
-		c.storeMounts()
-	}
-
+	// Pod is started
 	if err := p.startSetStates(); err != nil {
 		return err
 	}
 
+	for _, c := range p.containers {
+		if err := c.start(); err != nil {
+			return err
+		}
+	}
+
```

Comment:
So you're changing the container starting and state setting logic here, by moving it outside of hyperstart into pod.go. I like it, but it should be part of a separate patch on this PR.

Reply:
sure, @sameo at some point, I dont know if this is the correct time but I think we need to change the pod.start method to only start the pod ( through the agent ) and start the `pause` container, the rest of containers must be started by `StartContaine`

Reply:
That's almost what I'm asking you to do in a separate patch, except for separating the pause container from the other containers.

Any reason why we'd need to start the pause one at pod starting time ?

Reply:
@sameo I spitted the commit. I hope I helps for readability .
`Any reason why we'd need to start the pause one at pod starting time ?`
 it is not something I want but we already do, https://github.com/containers/virtcontainers/blob/master/hyperstart.go#L453, notice that this PR does not move that line to pod level. I dont see anything in the agent interface to to something like `h.stopPauseContainer`, so I am guessing that is hyperstart specific ? .